### PR TITLE
rpm: fix shared library devel package names and dependencies

### PR DESCRIPTION
**Pull request by smithfarm**
Fixes: http://tracker.ceph.com/issues/16345
Fixes: http://tracker.ceph.com/issues/16346

### CODE REVIEW

File: ceph.spec.in [hunk 1]
```diff
@@ -410,14 +413,15 @@ Striping interface built on top of the rados library, allowing
 to stripe bigger objects onto several standard rados objects using
 an interface very similar to the rados one.
 
-%package -n libradosstriper1-devel
+%package -n libradosstriper-devel
 Summary:	RADOS striping interface headers
 Group:		Development/Libraries
 License:	LGPL-2.0
 Requires:	libradosstriper1 = %{epoch}:%{version}-%{release}
 Requires:	librados2-devel = %{epoch}:%{version}-%{release}
```
Comment (tserong) [inline ceph.spec.in]: Should be `Requires: librados-devel`

File: ceph.spec.in [hunk 2]
```diff
@@ -471,14 +476,15 @@ performance, reliability, and scalability. This is a shared library
 allowing applications to access a Ceph distributed file system via a
 POSIX-like interface.
 
-%package -n libcephfs1-devel
+%package -n libcephfs-devel
 Summary:	Ceph distributed file system headers
 Group:		Development/Libraries
 License:	LGPL-2.0
 Requires:	libcephfs1 = %{epoch}:%{version}-%{release}
 Requires:	librados2-devel = %{epoch}:%{version}-%{release}
```
Comment (tserong) [inline ceph.spec.in]: Should be `Requires: librados-devel`

**Comment by tserong** [comment]
In addition to those two line notes, I did a quick search through the spec file, and I think the librbd-devel and devel-compat packages still also have `Requires: lib$FOO$VERSION-devel` lines, which should become `Requires: lib$FOO-devel` to match the renamed devel packages.

File: ceph.spec.in (hunk 1 above)
Comment (smithfarm) [inline ceph.spec.in]: fixed

**Comment by smithfarm** [comment]
Thanks, @tserong, for the review. I have made the necessary corrections.

@ktdreyer @b-ranto Your thoughts on this PR?

**Comment by smithfarm** [comment]
I'm trying to get the latest jewel into openSUSE "Tumbleweed". This appears to be the last obstacle...

**Comment by smithfarm** [comment]
@ktdreyer Done. Just `devel-compat`? Not any of the other "compats"?

**Comment by ktdreyer** [comment]
I think the other -compats could go as well, but that could be a separate PR :)

**Comment by ktdreyer** [comment]
`Reviewed-by: Ken Dreyer <kdreyer@redhat.com>`

**Comment by smithfarm** [comment]
pushed `wip-16345` to gitbuilders

**Comment by smithfarm** [comment]
gitbuilder likes it

**Comment by smithfarm** [comment]
Changelog:
* rebase
* add 0d9506a which ensures the old-named packages are provided as well as being obsoleted - as per https://en.opensuse.org/openSUSE:Package_dependencies#Renaming_a_package

**Comment by b-ranto** [comment]
This lgtm, I'd like to see this go through some upgrade path tests though.

**Comment by b-ranto** [comment]
Also, when I think about it, this could probably break a lot of (qa) tooling. How would ceph-deploy/ansible handle this?

**Comment by ktdreyer** [comment]
I checked ceph-deploy and ceph-ansible, and they don't bother installing any -devel RPMs.

**Comment by smithfarm** [comment]
Well, I've tried but these cleanups do not pass `upgrade/hammer-x` because it does

```
sudo yum -y install libcephfs_jni1 rbd-fuse ceph-radosgw librbd1 ceph-fuse python-ceph ceph ceph-devel ceph-test librados2 cephfs-java libcephfs1
```

(notice the presence of `ceph-devel`)

Next I will try `upgrade/infernalis-x` and `upgrade/jewel-x`.

**Comment by ktdreyer** [comment]
At this point, since Firefly is EOL, that test should probably be changed to use the -devel names from Hammer

**Comment by smithfarm** [comment]
JFYI `ceph-devel` has been dropped from the list of packages that get installed by teuthology - see https://github.com/ceph/ceph-qa-suite/pull/1068

**Comment by smithfarm** [comment]
Rebased again.

**Comment by smithfarm** [comment]
Repushed `wip-16345` to gitbuilders.

**Comment by smithfarm** [comment]
This passed gitbuilder but there's no point in running any more tests until https://github.com/ceph/teuthology/pull/912 is merged.

**Comment by smithfarm** [comment]
Hey Jenkins, test this please (logs are gone)

**Comment by smithfarm** [comment]
rebased, pushed to gitbuilder

**Comment by smithfarm** [comment]
**running** http://pulpito.ceph.com/smithfarm-2016-08-08_02:50:55-upgrade:jewel-x-wip-16345---basic-smithi/

**Comment by smithfarm** [comment]
**fail** http://pulpito.ceph.com/smithfarm-2016-08-08_05:24:13-upgrade:jewel-x:point-to-point-x-wip-16345---basic-smithi/

Unrelated failure. The actual upgrade goes fine.

```
2016-08-08T07:20:06.916 INFO:tasks.workunit.client.1.smithi015.stdout:test/cls_rbd/test_cls_rbd.cc:1384: Failure
2016-08-08T07:20:06.916 INFO:tasks.workunit.client.1.smithi015.stdout:Value of: mirror_image_set(&ioctx, "image_id2", image2)
2016-08-08T07:20:06.916 INFO:tasks.workunit.client.1.smithi015.stdout:  Actual: -2
2016-08-08T07:20:06.917 INFO:tasks.workunit.client.1.smithi015.stdout:Expected: 0
2016-08-08T07:20:06.998 INFO:tasks.workunit.client.1.smithi015.stdout:[  FAILED  ] TestClsRbd.mirror_image (5 ms)
```

**Comment by smithfarm** [comment]
The jewel-x upgrade suite is in sorry shape.

**Comment by smithfarm** [comment]
**running** http://pulpito.ceph.com/smithfarm-2016-08-08_07:44:47-upgrade:jewel-x-wip-16345---basic-smithi/

Chose one that currently succeeds in the nightlies.

**Comment by smithfarm** [comment]
@ktdreyer @b-ranto The upgrade/jewel-x job passed. OK to merge?

**Comment by ktdreyer** [comment]
:+1: thanks

**Comment by b-ranto** [comment]
:+1:

**Pull request merged**